### ISSUE TITLE
Formatting svelte files

### ISSUE THREAD

**Issue by agorer** [jscheid/prettier.el#88]
**Describe the bug**
When formatting a svelte files it seems like it thinks that it is a js file. Giving errors like 

```
/Users/.../src/pages/_layout.svelte: SyntaxError: Private identifiers are not allowed outside class bodies. (9:2)
   7 | </script>
   8 |
>  9 | {#if $user.isAuthenticated}
     |  ^
  10 |   <div class="containter">
  11 |     <div>
  12 |       <slot />
```

**Link to `M-x prettier-info` output**
https://gist.github.com/agorer/c57f881d75d39779a1a91a71b4f8bf15

**To Reproduce**
1. Install svelte plugin for Prettier (https://github.com/sveltejs/prettier-plugin-svelte).
2. Create a svelte file like:
```
<script lang="ts">
  import { goto } from '@roxi/routify'
  import Footer from 'layout/Footer.svelte'
  import { user } from "auth/store"

</script>

{#if $user.isAuthenticated}
  <div class="containter">
    <div>
      <slot />
    </div>

    <Footer />
  </div>
{:else}
  {$goto('/auth/login')}
{/if}
```
3. Try to format the file within emacs.

**Expected behavior**
The file should be formatted according to the rules in .prettierrc

**Additional context**

**Comment by jscheid** [comment]
This is due to this package deriving the Prettier parser from the major mode instead of letting Prettier choose the parser.

It's something I've been meaning to change, I'll see that I can push an improvement for this soon.

As a workaround, if you use [`svelte-mode`](https://melpa.org/#/svelte-mode) it should use the correct parser but I can see why you would want to use `typescript-mode` instead.

**Comment by agorer** [comment]
> It's something I've been meaning to change, I'll see that I can push an improvement for this soon.
Thanks, looking forward to it.

> As a workaround, if you use [`svelte-mode`](https://melpa.org/#/svelte-mode) it should use the correct parser but I can see why you would want to use `typescript-mode` instead.
I am using svelte-mode, but it changes the major mode depending on where you are in the file (ts / js mode inside the script tags, svelte in the markup and css within the style tags).

**Comment by jscheid** [comment]
> I am using svelte-mode

Right, that's different. I've pushed a fix, it should have landed in `20210606.1152`. Can you let me know if it works better?

**Comment by agorer** [comment]
Yes, it is working now anywhere in the file, thank you!
